src/TimeTravel.elm

```diff
@@ -3,3 +3,104 @@ module TimeTravel exposing (addTimeTravel)
 import Playground exposing (..)
 import Set
 
+controlBarHeight = 64
+maxVisibleHistory = 2000
+
+addTimeTravel rawGame = 
+  { initialState = initialStateWithTimeTravel rawGame
+  , updateState = updateWithTimeTravel rawGame
+  , view = viewWithTimeTravel rawGame
+  }
+
+initialStateWithTimeTravel rawGame = 
+  { rawModel = rawGame.initialState
+  , paused = False
+  , history = []
+  , historyPlaybackPosition = 0
+  }
+
+updateWithTimeTravel rawGame computer currentModel =
+  let
+    updatedModel = 
+      { currentModel
+      | rawModel = rawGame.updateState computer currentModel.rawModel
+      , history = currentModel.history ++ [computer]
+      , historyPlaybackPosition = currentModel.historyPlaybackPosition + 1
+      }
+    newPlaybackPosition = 
+      min
+        (mousePosToHistoryIndex computer)
+        (List.length currentModel.history)
+    replayHistory pastInputs = 
+      List.foldl 
+        (\input model -> rawGame.updateState input model)
```

Comment:
I have no idea why I couldn't think of currying and instead made a weird lambda function. Of course it makes more sense to do rawGame.updateState instead of this whole thing. It still works, but it's bad.